CONTRIBUTING.md

```diff
@@ -29,6 +29,69 @@ Documentation for the in-the-wild deployments lives in the
+
+**Examples of casual contribution**
+- Adjustments to documentation
+- Adjustments to configuration parameters
+- Fixing a script or otherwise non-mission critical piece of code
+
+**Process**
+
+1. Fork the Wormhole project ([GitHub Official Documentation](https://docs.github.com/en/get-started/quickstart/fork-a-repo)).
+1. Clone the forked repository ([GitHub Official Documentation](https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository#cloning-a-repository))
+1. Change directory into the Wormhole repository
+    - `cd ./wormhole`
+1. Create a new branch
+    - `git checkout -b NAME_OF_FEATURE_BRANCH`
+1. Make changes to relevant files
+1. Add your changes
+      - `git add FILE_YOU_CHANGED`
+1. Commit your changes
+      - `git commit -m "docs: updated contributing process"`
+      - *See below for commit message convention guidance*
+1. Push your changes to GitHub
+      - `git push origin NAME_OF_FEATURE_BRANCH"`
+1. On the GitHub PR, Propose 2 core contributors -OR- Wait for 2 core contributors to review and provide feedback.
+1. Iterate with the reviewers to ensure that the necessary changes are made and once all changes are made and reviews are complete a core contributor will merge the code into the dev.v2 branch.
+
+### Gerrit Change Set (aka: Advanced Contribution)
+**Objective:** Allow advanced contributors, often core contributors, to make major changes to the project using a more advanced review toolchain, allowing for more efficient and iterative change sets and preventing re-reviews of code, which has already been reviewed.
+
+**Examples of Advanced contribution**
+- Adding a new chain integration
+- Adding a new core feature to the protocol
+- Initial draft of a protocol specification/whitepaper
+- Any non-config changes to the Guardian/Smart Contracts
+
+**Process**
+
+1. Fork the Wormhole project ([GitHub Official Documentation](https://docs.github.com/en/get-started/quickstart/fork-a-repo))
+1. INSERT_STEPS_TO_GET_SETUP_USING_GERRIT_POSSIBLY_ADDING_A_GERRIT_INSTANCE_FOR_WORMHOLE
+1. Copy the "Clone with commit-msg hook" command from Gerrit to pull down the repository.
+1. Change directory into the Wormhole repository
+    - `cd ./wormhole`
+1. Create a new branch
+    - `git checkout -b NAME_OF_FEATURE_BRANCH`
+1. Make changes to relevant files
+1. Add your changes
+      - `git add FILE_YOU_CHANGED`
+1. Commit your changes
+      - `git commit -m "docs: updated contributing process"`
+      - *See below for commit message convention guidance*
+1. Push your changes to Gerrit
+      - `git push origin HEAD:refs/for/main"`
+1. On the Gerrit Change Set, Propose 2 core contributors -OR- Wait for 2 core contributors to review, provided feedback, and approve your code.
+1. Iterate with the reviewers to ensure that the necessary changes are made and once all changes are made and reviews are complete a core contributor will merge the code into the dev.v2 branch.
```

Comment:
Might want some better instruction on the nuance for how changesets are reviewed in Gerrit.

Reply:
Not sure if you're referring to general code review or the specific mechanics of how to address feedback in gerrit.  

For general code review, I mentioned below that the [carbon-lang docs](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/code_review.md) have a good overview of what to expect out of the code review process.  

For gerrit-specific things I think it would be good to mention `git commit --fixup` and `git rebase --autosquash` here.  For selectively staging some parts of a file for a fixup commit, we can suggest:

  * [magit](https://magit.vc/) for emacs users
  * [fugitive](https://github.com/tpope/vim-fugitive) for vim/nvim users
  * [Stage selected ranges](https://code.visualstudio.com/docs/getstarted/tips-and-tricks#_staging) for vscode
  * [gitui](https://github.com/extrawurst/gitui) for an editor-agnostic option

There is also [git-autofixup](https://github.com/torbiak/git-autofixup/), which can automate some of this.